browser/brave_vpn/dns/brave_vpn_dns_observer_factory.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/dns/brave_vpn_dns_observer_factory.h"
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "base/feature_list.h"
+#include "brave/browser/brave_vpn/dns/brave_vpn_dns_observer_service.h"
+#include "brave/components/brave_vpn/brave_vpn_utils.h"
+#include "brave/components/brave_vpn/features.h"
+#include "brave/components/brave_vpn/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/policy/chrome_browser_policy_connector.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/policy/core/common/cloud/cloud_policy_store.h"
+#include "components/policy/core/common/cloud/machine_level_user_cloud_policy_manager.h"
+#include "components/policy/core/common/cloud/machine_level_user_cloud_policy_store.h"
+#include "components/policy/core/common/policy_map.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+
+namespace brave_vpn {
+
+namespace {
+absl::optional<std::string> GetPolicyValue(const std::string& name) {
```

Comment:
we don't need this anymore, as discussed in dm we are checking the force managed property in the config reader

Reply:
```bool is_managed =
      local_state_->FindPreference(prefs::kDnsOverHttpsMode)->IsManaged();```

Reply:
actually we don't need this because we should be calling `GetSecureDnsConfiguration` instead of `UpdateNetworkService` and then we just check the result to see if we did the override or not in the  config reader

Reply:
we can call it before setting the pref to get the current values and then call it after to trigger the update. That's why all the logic for deciding whether to override was supposed to go in the config reader.

Reply:
this function was removed